### ISSUE TITLE
Using auipc + jalr when jump to a far label.

### ISSUE THREAD

**Issue by luyahan** [v8-riscv/v8#183]
**Is your feature request related to a problem? Please describe.**
When i fix #142 #87 . I found offset(1048908) between  label and j instr exceeding range of `jal` (kMaxJumpOffset=1048575).
But TrampolinePool is designed for branch instr and only emmit once.
In #87 it is emmitd at pos:3908 after kMaxBranchOffset(https://github.com/v8-riscv/v8/blob/riscv64/src/codegen/riscv64/assembler-riscv64.cc#L215)
```
inserting trampoline pool at 0x562a0ea89b84 (3908),num:36
```

**Describe the solution you'd like**
using auipc + jalr when jump a unbound label.

**Describe alternatives you've considered**
When `j` a unbound label, load a unbound label address  and `jalr  x0,t3,0x0`
or Add another TrampolinePool designed for jal
or jsut skip the case like ppc:
```
PPC/AIX : skip regex test case
Regex test cases fails on AIX debug due to stack-overflow
```
infer to https://chromium-review.googlesource.com/c/v8/v8/+/1539791

**Additional context**
None

**Comment by luyahan** [comment]
I had implement jump unbound label by auipc  + jalr in PLCT repo.

**Comment by penguinwu** [comment]
Unbound label typically means that a label is not defined in generated codes (i.e., not specified in any `bind(&x)`). From your description, it seems that when you say "unbound" label, do you mean that the label is too far away from the current PC to be represented in the 20-bit immediate field of `jal`? 

Could you confirm which case you refer to?

**Comment by luyahan** [comment]
> Unbound label typically means that a label is not defined in generated codes (i.e., not specified in any `bind(&x)`). From your description, it seems that when you say "unbound" label, do you mean that the label is too far away from the current PC to be represented in the 20-bit immediate field of `jal`? 
> 
> Could you confirm which case you refer to?

Yes, it so far . I add 'lui' in assembler:bind_to like branch, run case in #142 occurs check failed.

**Comment by luyahan** [comment]
> Unbound label typically means that a label is not defined in generated codes (i.e., not specified in any `bind(&x)`). From your description, it seems that when you say "unbound" label, do you mean that the label is too far away from the current PC to be represented in the 20-bit immediate field of `jal`? 
> 
> Could you confirm which case you refer to?

If jump a unbound label，it emmit a j instr. The offset of j will be set when the label is binded.But the case in #142，the addr of bind label is too far away from addr of jump instr ecceeding max jump offset.

**Comment by luyahan** [comment]
> Unbound label typically means that a label is not defined in generated codes (i.e., not specified in any `bind(&x)`). From your description, it seems that when you say "unbound" label, do you mean that the label is too far away from the current PC to be represented in the 20-bit immediate field of `jal`? 
> 
> Could you confirm which case you refer to?
 This cas:
'''
​out/riscv64.sim/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-1067270.js --random-seed=-164473607 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --allow-natives-syntax
'''

**Comment by penguinwu** [comment]
> I had implement jump unbound label by auipc + jalr in PLCT repo.

@luyahan Your suggested fix of auipc + jalr makes sense to me. Do you want to port your implementation to our repo?

**Comment by luyahan** [comment]
> > I had implement jump unbound label by auipc + jalr in PLCT repo.
> 
> @luyahan Your suggested fix of auipc + jalr makes sense to me. Do you want to port your implementation to our repo?

Yes, it dones't alter too much code .And we can add a flag to control it.If user encounter this error , remind user to add the flag.

We can continue to optimize tis later.

**Comment by penguinwu** [comment]
Why do we need a flag to control it? Since the jump offset is determined at compile-time, if an offset is beyond the range `imm20`, we can always generate `auipc + jalr` because without the fix the code would not function correctly.

**Comment by luyahan** [comment]
> Why do we need a flag to control it? Since the jump offset is determined at compile-time, if an offset is beyond the range `imm20`, we can always generate `auipc + jalr` because without the fix the code would not function correctly.
If jump a binded label , we can select instr accord to jump offset.
But on a extreme case, emit  a `j` instr because of a unbound label, don't known the jump offset which may be beyond `imm20`.

**Comment by luyahan** [comment]
> Why do we need a flag to control it? Since the jump offset is determined at compile-time, if an offset is beyond the range `imm20`, we can always generate `auipc + jalr` because without the fix the code would not function correctly.

I decide to alter tramplinepool. Alter `j` instr in tramplinepool into `auipc` + `jalr`,so it can fix #142 #87

**Comment by luyahan** [comment]
Pr #189
